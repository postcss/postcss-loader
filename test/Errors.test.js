'use strict'

const webpack = require('./helpers/compiler')
const { loader } = require('./helpers/compilation')

describe('Errors', () => {
  test('Validation Error', () => {
    const loader = require('../lib')

    const error = () => loader.call({ query: { sourceMap: 1 } })

    expect(error).toThrow()
    expect(error).toThrowErrorMatchingSnapshot()
  })

  test('Syntax Error', () => {
    const config = {
      loader: {
        options: {
          parser: 'sugarss'
        }
      }
    }

    return webpack('css/index.js', config).then((stats) => {
      const error = loader(stats).err

      expect(error[0]).toMatchSnapshot()
    })
  })
})
