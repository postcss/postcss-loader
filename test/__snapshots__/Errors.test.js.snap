// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Syntax Error 1`] = `
"Syntax Error 

(1:3) Unexpected separator in property

[31m[1m>[22m[39m[90m 1 | [39ma [33m{[39m color[33m:[39m black [33m}[39m
 [90m   | [39m  [31m[1m^[22m[39m
 [90m 2 | [39m
"
`;

exports[`Errors Validation Error 1`] = `
"ValidationError: PostCSS Loader Invalid Options

options.sourceMap should be string,boolean
"
`;
